### ISSUE TITLE
Addition with dfm objects drops attributes

### ISSUE THREAD

**Issue by chmue** [quanteda/quanteda#1279]
Continuation from as discussion in #1276 

The addition operation for dfms resets all the attributes. The offending part is `as(x, "dgCMatrix")` which **removes all dfm-related slots** that are not used in "dgCMatrix" in [this line](https://github.com/quanteda/quanteda/blob/master/R/dfm-classes.R#L105):

```r
str(as(data_dfm_lbgexample, "dgCMatrix"))
```
```
Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  ..@ i       : int [1:102] 0 0 0 0 0 0 1 0 1 0 ...
  ..@ p       : int [1:38] 0 1 2 3 4 5 7 9 12 15 ...
  ..@ Dim     : int [1:2] 6 37
  ..@ Dimnames:List of 2
  .. ..$ docs    : chr [1:6] "R1" "R2" "R3" "R4" ...
  .. ..$ features: chr [1:37] "A" "B" "C" "D" ...
  ..@ x       : num [1:102] 2 3 10 22 45 78 2 115 3 146 ...
  ..@ factors : list()
```

**Comment by chmue** [comment]
There is `reassign_slots` in `utils.R` which I will try to use fix the error in this and possibly other methods which do some operation on the matrix and return a dfm.

**Comment by kbenoit** [comment]
Feel free to fix this in PR #1276.

**Comment by chmue** [comment]
Ok, will do!

**Comment by koheiw** [comment]
solved along with #1274
